FindMKL.cmake

```diff
@@ -0,0 +1,222 @@
+#[=======================================================================[.rst:
+FindMKL
+-------
+
+Example usage
+^^^^^^^^^^^^^
+
+  find_package(MKL REQUIRED OPTIONAL_COMPONENTS SCALAPACK)
+  target_link_libraries(... MKL::MKL)
```

Comment:
What will our convention be for target names? I think, all uppercase is not very common.